src/core/hash.zig

```diff
@@ -46,4 +49,13 @@ pub const Hash = struct {
         }
         return CompareResult.Equal;
     }
+
+    pub fn extend_and_hash(id: Hash, val: []u8, alloc: Allocator) !Self {
```

Comment:
can you make the errors explicit - ie, this fcn should be `error { OutOfMemory }` since thats the only error it returns